app/services/rate_creator.rb

```diff
@@ -13,17 +14,17 @@ def initialize(value, submission, user, rate_checker)
   end
 
   def call
-    rate = Rate.new({ value: @value, submission: @submission, user: @user })
-
-    if @rate_checker.user_has_already_rated?(@submission.id, @user.id)
-        errors = ['You are not allowed to rate the same submission twice']
-        success = false
-        result = Result.new(rate, success, errors)
+    if rate_checker.user_has_already_rated?
```

Comment:
And if there is not other use of the method we could delete it from the `RateChecker`